services/tripService.js

```diff
@@ -57,3 +57,17 @@ export async function deleteTrip(id) {
 
   return deletedTrip;
 }
+
+export async function deleteUserAccount(userEmail) {
+  await dbConnect();
+
+  const trips = await Trip.find({ userEmail: userEmail });
+  if (!Array.isArray(trips)) throw new Error();
```

Comment:
You can pass in a descriptive message inside of `Error()` for better developer experience (DX).

Reply:
The error message (500 - internal server error) is thrown in the api catch. I throw this error here when there's an unexpected return (not array).